File: lib/screens/centro_acopio/centro_acopio_home_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
// import 'package:cloud_firestore/cloud_firestore.dart'; // TODO: Descomentar para producción
// import 'package:firebase_auth/firebase_auth.dart'; // TODO: Descomentar para producción
import '../../utils/colors.dart';
import 'recepcion_material_screen.dart';
import 'inventario_screen.dart';
import 'reportes_screen.dart';
import 'prepago_screen.dart';

class CentroAcopioHomeScreen extends StatefulWidget {
  const CentroAcopioHomeScreen({super.key});

  @override
  State<CentroAcopioHomeScreen> createState() => _CentroAcopioHomeScreenState();
}

class _CentroAcopioHomeScreenState extends State<CentroAcopioHomeScreen> {
  // MODO DISEÑO: Datos de prueba
  final Map<String, dynamic> _centroAcopio = {
    'nombre': 'Centro de Acopio BioWay',
    'direccion': 'Av. Tecnológico #100',
    'municipio': 'Aguascalientes',
    'estado': 'Aguascalientes',
    'codigoPostal': '20100',
    'saldoPrepago': 15420.50,
    'reputacion': 4.8,
    'totalRecepcionesMes': 156,
    'kgRecibidosHoy': 450.5,
    'kgRecibidosMes': 12340.8,
  };

  final List<Map<String, dynamic>> _ultimasRecepciones = [
    {
      'brindador': 'María García',
      'material': 'PET Tipo 1',
      'cantidad': 15.5,
      'hora': '10:30 AM',
      'pago': 77.50,
    },
    {
      'brindador': 'Juan Pérez',
      'material': 'Cartón',
      'cantidad': 25.0,
      'hora': '09:15 AM',
      'pago': 50.00,
    },
    {
      'brindador': 'Ana López',
      'material': 'Vidrio',
      'cantidad': 8.3,
      'hora': '08:45 AM',
      'pago': 24.90,
    },
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: BioWayColors.backgroundGrey,
      body: SafeArea(
        child: CustomScrollView(
          slivers: [
            // AppBar personalizado
            SliverAppBar(
              expandedHeight: 180,
              floating: false,
              pinned: true,
              backgroundColor: BioWayColors.primaryGreen,
              flexibleSpace: FlexibleSpaceBar(
                title: const Text(
                  'Centro de Acopio',
                  style: TextStyle(
                    color: Colors.white,
                    fontWeight: FontWeight.bold,
                    fontSize: 20,
                  ),
                ),
                background: Container(
                  decoration: BoxDecoration(
                    gradient: LinearGradient(
                      begin: Alignment.topLeft,
                      end: Alignment.bottomRight,
                      colors: [
                        BioWayColors.primaryGreen,
                        BioWayColors.mediumGreen,
                      ],
                    ),
                  ),
                  child: Stack(
                    children: [
                      Positioned(
                        right: -50,
                        top: -50,
                        child: Container(
                          width: 200,
                          height: 200,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: Colors.white.withOpacity(0.1),
                          ),
                        ),
                      ),
                      Positioned(
                        left: -30,
                        bottom: -30,
                        child: Container(
                          width: 150,
                          height: 150,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: Colors.white.withOpacity(0.05),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
              leading: IconButton(
                icon: const Icon(Icons.arrow_back, color: Colors.white),
                onPressed: () => Navigator.pop(context),
              ),
              actions: [
                IconButton(
                  icon: const Icon(Icons.notifications_outlined, color: Colors.white),
                  onPressed: () {
                    ScaffoldMessenger.of(context).showSnackBar(
                      SnackBar(
                        content: const Text('Notificaciones (próximamente)'),
                        backgroundColor: BioWayColors.info,
                      ),
                    );
                  },
                ),
                IconButton(
                  icon: const Icon(Icons.settings_outlined, color: Colors.white),
                  onPressed: () {
                    ScaffoldMessenger.of(context).showSnackBar(
                      SnackBar(
                        content: const Text('Configuración (próximamente)'),
                        backgroundColor: BioWayColors.info,
                      ),
                    );
                  },
                ),
              ],
            ),
            
            // Contenido
            SliverToBoxAdapter(
              child: Padding(
                padding: const EdgeInsets.all(16),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    // Información del centro
                    _buildInfoCard(),
                    const SizedBox(height: 20),
                    
                    // Estadísticas rápidas
                    _buildQuickStats(),
                    const SizedBox(height: 20),
                    
                    // Menú principal
                    Text(
                      'Operaciones',
                      style: TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                        color: BioWayColors.darkGreen,
                      ),
                    ),
                    const SizedBox(height: 12),
                    _buildMenuGrid(),
                    const SizedBox(height: 20),
                    
                    // Últimas recepciones
                    Text(
                      'Últimas Recepciones',
                      style: TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                        color: BioWayColors.darkGreen,
                      ),
                    ),
                    const SizedBox(height: 12),
                    _buildRecentReceptions(),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton.extended(
        onPressed: _showQRScanner,
        backgroundColor: BioWayColors.primaryGreen,
        icon: const Icon(Icons.qr_code_scanner, color: Colors.white),
        label: const Text(
          'Escanear QR',
          style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold),
        ),
      ),
    );
  }

  Widget _buildInfoCard() {
    return Card(
      elevation: 4,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
      ),
      child: Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(16),
          gradient: LinearGradient(
            begin: Alignment.topLeft,
            end: Alignment.bottomRight,
            colors: [
              Colors.white,
              BioWayColors.primaryGreen.withOpacity(0.05),
            ],
          ),
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              children: [
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: BioWayColors.primaryGreen.withOpacity(0.1),
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Icon(
                    Icons.warehouse,
                    color: BioWayColors.primaryGreen,
                    size: 28,
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        _centroAcopio['nombre'],
                        style: const TextStyle(
                          fontSize: 18,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      const SizedBox(height: 4),
                      Row(
                        children: [
                          Icon(Icons.location_on, size: 14, color: Colors.grey.shade600),
                          const SizedBox(width: 4),
                          Expanded(
                            child: Text(
                              '${_centroAcopio['direccion']}, ${_centroAcopio['municipio']}',
                              style: TextStyle(
                                fontSize: 12,
                                color: Colors.grey.shade600,
                              ),
                              overflow: TextOverflow.ellipsis,
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                ),
              ],
            ),
            const SizedBox(height: 16),
            Container(
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: BioWayColors.success.withOpacity(0.1),
                borderRadius: BorderRadius.circular(12),
                border: Border.all(
                  color: BioWayColors.success.withOpacity(0.3),
                ),
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        'Saldo Prepago',
                        style: TextStyle(
                          fontSize: 12,
                          color: Colors.grey.shade600,
                        ),
                      ),
                      const SizedBox(height: 4),
                      Text(
                        '\$${_centroAcopio['saldoPrepago'].toStringAsFixed(2)}',
                        style: TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                          color: BioWayColors.success,
                        ),
                      ),
                    ],
                  ),
                  ElevatedButton.icon(
                    onPressed: () {
                      Navigator.push(
                        context,
                        MaterialPageRoute(
                          builder: (context) => const PrepagoScreen(),
                        ),
                      );
                    },
                    icon: const Icon(Icons.add, size: 18),
                    label: const Text('Recargar'),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: BioWayColors.success,
                      foregroundColor: Colors.white,
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(20),
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildQuickStats() {
    return Row(
      children: [
        Expanded(
          child: _buildStatCard(
            'Recepciones Hoy',
            '${_ultimasRecepciones.length}',
            Icons.today,
            BioWayColors.info,
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: _buildStatCard(
            'Kg Hoy',
            '${_centroAcopio['kgRecibidosHoy']}',
            Icons.scale,
            BioWayColors.warning,
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: _buildStatCard(
            'Reputación',
            '${_centroAcopio['reputacion']}',
            Icons.star,
            BioWayColors.success,
          ),
        ),
      ],
    );
  }

  Widget _buildStatCard(String title, String value, IconData icon, Color color) {
    return Container(
      padding: const EdgeInsets.all(12),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.05),
            blurRadius: 10,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Column(
        children: [
          Icon(icon, color: color, size: 24),
          const SizedBox(height: 8),
          Text(
            value,
            style: TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.bold,
              color: color,
            ),
          ),
          const SizedBox(height: 4),
          Text(
            title,
            style: TextStyle(
              fontSize: 11,
              color: Colors.grey.shade600,
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }

  Widget _buildMenuGrid() {
    final menuItems = [
      {
        'title': 'Recepción',
        'icon': Icons.inbox,
        'color': BioWayColors.primaryGreen,
        'screen': const RecepcionMaterialScreen(),
      },
      {
        'title': 'Inventario',
        'icon': Icons.inventory_2,
        'color': BioWayColors.info,
        'screen': const InventarioScreen(),
      },
      {
        'title': 'Reportes',
        'icon': Icons.analytics,
        'color': BioWayColors.warning,
        'screen': const ReportesScreen(),
      },
      {
        'title': 'Prepago',
        'icon': Icons.account_balance_wallet,
        'color': BioWayColors.success,
        'screen': const PrepagoScreen(),
      },
    ];

    return GridView.builder(
      shrinkWrap: true,
      physics: const NeverScrollableScrollPhysics(),
      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
        crossAxisCount: 2,
        childAspectRatio: 1.5,
        crossAxisSpacing: 12,
        mainAxisSpacing: 12,
      ),
      itemCount: menuItems.length,
      itemBuilder: (context, index) {
        final item = menuItems[index];
        return InkWell(
          onTap: () {
            HapticFeedback.lightImpact();
            Navigator.push(
              context,
              MaterialPageRoute(
                builder: (context) => item['screen'] as Widget,
              ),
            );
          },
          borderRadius: BorderRadius.circular(16),
          child: Container(
            decoration: BoxDecoration(
              color: Colors.white,
              borderRadius: BorderRadius.circular(16),
              boxShadow: [
                BoxShadow(
                  color: Colors.black.withOpacity(0.05),
                  blurRadius: 10,
                  offset: const Offset(0, 2),
                ),
              ],
            ),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: (item['color'] as Color).withOpacity(0.1),
                    shape: BoxShape.circle,
                  ),
                  child: Icon(
                    item['icon'] as IconData,
                    color: item['color'] as Color,
                    size: 28,
                  ),
                ),
                const SizedBox(height: 8),
                Text(
                  item['title'] as String,
                  style: const TextStyle(
                    fontSize: 14,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ],
            ),
          ),
        );
      },
    );
  }

  Widget _buildRecentReceptions() {
    return Column(
      children: _ultimasRecepciones.map((recepcion) {
        return Card(
          margin: const EdgeInsets.only(bottom: 12),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
          ),
          child: ListTile(
            leading: CircleAvatar(
              backgroundColor: BioWayColors.primaryGreen.withOpacity(0.1),
              child: Icon(
                Icons.person,
                color: BioWayColors.primaryGreen,
              ),
            ),
            title: Text(
              recepcion['brindador'],
              style: const TextStyle(fontWeight: FontWeight.bold),
            ),
            subtitle: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const SizedBox(height: 4),
                Row(
                  children: [
                    Icon(Icons.recycling, size: 14, color: Colors.grey.shade600),
                    const SizedBox(width: 4),
                    Text(
                      '${recepcion['material']} - ${recepcion['cantidad']} kg',
                      style: TextStyle(fontSize: 12, color: Colors.grey.shade600),
                    ),
                  ],
                ),
                const SizedBox(height: 2),
                Row(
                  children: [
                    Icon(Icons.schedule, size: 14, color: Colors.grey.shade600),
                    const SizedBox(width: 4),
                    Text(
                      recepcion['hora'],
                      style: TextStyle(fontSize: 12, color: Colors.grey.shade600),
                    ),
                  ],
                ),
              ],
            ),
            trailing: Container(
              padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
              decoration: BoxDecoration(
                color: BioWayColors.success.withOpacity(0.1),
                borderRadius: BorderRadius.circular(20),
              ),
              child: Text(
                '\$${recepcion['pago'].toStringAsFixed(2)}',
                style: TextStyle(
                  color: BioWayColors.success,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ),
        );
      }).toList(),
    );
  }

  void _showQRScanner() {
    showModalBottomSheet(
      context: context,
      isScrollControlled: true,
      backgroundColor: Colors.transparent,
      builder: (context) => Container(
        height: MediaQuery.of(context).size.height * 0.8,
        decoration: const BoxDecoration(
          color: Colors.white,
          borderRadius: BorderRadius.only(
            topLeft: Radius.circular(24),
            topRight: Radius.circular(24),
          ),
        ),
        child: Column(
          children: [
            Container(
              margin: const EdgeInsets.only(top: 12),
              width: 40,
              height: 4,
              decoration: BoxDecoration(
                color: Colors.grey.shade300,
                borderRadius: BorderRadius.circular(2),
              ),
            ),
            const SizedBox(height: 20),
            Text(
              'Escanear Código QR',
              style: TextStyle(
                fontSize: 20,
                fontWeight: FontWeight.bold,
                color: BioWayColors.darkGreen,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              'Coloca el código QR del brindador en el centro',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey.shade600,
              ),
            ),
            const SizedBox(height: 20),
            Expanded(
              child: Container(
                margin: const EdgeInsets.all(20),
                decoration: BoxDecoration(
                  borderRadius: BorderRadius.circular(20),
                  border: Border.all(
                    color: BioWayColors.primaryGreen,
                    width: 2,
                  ),
                ),
                child: Center(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Icon(
                        Icons.qr_code_scanner,
                        size: 100,
                        color: BioWayColors.primaryGreen.withOpacity(0.5),
                      ),
                      const SizedBox(height: 20),
                      Text(
                        '🎨 MODO DISEÑO',
                        style: TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.bold,
                          color: BioWayColors.warning,
                        ),
                      ),
                      const SizedBox(height: 8),
                      Text(
                        'Escáner QR deshabilitado',
                        style: TextStyle(
                          fontSize: 14,
                          color: Colors.grey.shade600,
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ),
            Padding(
              padding: const EdgeInsets.all(20),
              child: ElevatedButton(
                onPressed: () {
                  Navigator.pop(context);
                  Navigator.push(
                    context,
                    MaterialPageRoute(
                      builder: (context) => const RecepcionMaterialScreen(),
                    ),
                  );
                },
                style: ElevatedButton.styleFrom(
                  backgroundColor: BioWayColors.primaryGreen,
                  minimumSize: const Size(double.infinity, 50),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(25),
                  ),
                ),
                child: const Text(
                  'Ingresar Manual',
                  style: TextStyle(
                    color: Colors.white,
                    fontSize: 16,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/biowaymexico/ui/screens/centro_acopio/CentroAcopioHomeScreen.kt
package com.biowaymexico.ui.screens.centro_acopio

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavHostController
import com.biowaymexico.ui.theme.BioWayColors

/**
 * Pantalla principal del Centro de Acopio
 * Replicado fielmente del diseño Flutter original
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CentroAcopioHomeScreen(navController: NavHostController) {
    Scaffold(
        containerColor = Color(0xFFF5F5F5),
        topBar = {
            CentroAcopioTopBar()
        },
        floatingActionButton = {
            FloatingActionButton(
                onClick = { /* TODO: Escanear QR */ },
                containerColor = BioWayColors.NavGreen
            ) {
                Row(
                    modifier = Modifier.padding(horizontal = 16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.QrCodeScanner,
                        contentDescription = "Escanear QR",
                        tint = Color.White
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = "Escanear QR",
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                }
            }
        }
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(20.dp)
        ) {
            item { BuildInfoCard() }
            item { BuildQuickStats() }
            item {
                Text(
                    text = "Operaciones",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = BioWayColors.DarkGreen
                )
            }
            item { BuildMenuGrid() }
            item {
                Text(
                    text = "Últimas Recepciones",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = BioWayColors.DarkGreen
                )
            }
            item { BuildRecentReceptions() }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun CentroAcopioTopBar() {
    MediumTopAppBar(
        title = {
            Text(
                text = "Centro de Acopio",
                color = Color(0xFF00553F),
                fontWeight = FontWeight.Bold,
                fontSize = 20.sp
            )
        },
        colors = TopAppBarDefaults.mediumTopAppBarColors(
            containerColor = Color.Transparent
        ),
        modifier = Modifier.background(
            brush = Brush.linearGradient(
                colors = listOf(
                    BioWayColors.MediumGreen,
                    BioWayColors.AquaGreen
                )
            )
        )
    )
}

@Composable
private fun BuildInfoCard() {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        color = Color.White,
        shadowElevation = 4.dp
    ) {
        Column(
            modifier = Modifier.padding(20.dp)
        ) {
            Text(
                text = "Centro de Acopio BioWay",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = BioWayColors.DarkGreen
            )
            Spacer(modifier = Modifier.height(8.dp))
            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = Icons.Default.LocationOn,
                    contentDescription = null,
                    tint = Color(0xFF666666),
                    modifier = Modifier.size(18.dp)
                )
                Spacer(modifier = Modifier.width(4.dp))
                Text(
                    text = "Av. Tecnológico #100, Aguascalientes",
                    fontSize = 14.sp,
                    color = Color(0xFF666666)
                )
            }
            Spacer(modifier = Modifier.height(16.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                InfoItem("Saldo Prepago", "$15,420.50")
                InfoItem("Reputación", "4.8 ⭐")
            }
        }
    }
}

@Composable
private fun InfoItem(label: String, value: String) {
    Column {
        Text(
            text = label,
            fontSize = 12.sp,
            color = Color(0xFF999999)
        )
        Text(
            text = value,
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = BioWayColors.NavGreen
        )
    }
}

@Composable
private fun BuildQuickStats() {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        StatCard(
            icon = Icons.Default.Today,
            value = "450.5 kg",
            label = "Recibidos Hoy",
            color = BioWayColors.SuccessGreen,
            modifier = Modifier.weight(1f)
        )
        StatCard(
            icon = Icons.Default.CalendarMonth,
            value = "12,340 kg",
            label = "Este Mes",
            color = BioWayColors.InfoBlue,
            modifier = Modifier.weight(1f)
        )
    }
}

@Composable
private fun StatCard(
    icon: ImageVector,
    value: String,
    label: String,
    color: Color,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(16.dp),
        color = Color.White,
        shadowElevation = 4.dp
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = color,
                modifier = Modifier.size(32.dp)
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = value,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = BioWayColors.DarkGreen
            )
            Text(
                text = label,
                fontSize = 12.sp,
                color = Color(0xFF999999)
            )
        }
    }
}

@Composable
private fun BuildMenuGrid() {
    Column(
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            MenuCard(
                icon = Icons.Default.AddCircle,
                title = "Recepción",
                color = BioWayColors.SuccessGreen,
                onClick = { /* TODO */ },
                modifier = Modifier.weight(1f)
            )
            MenuCard(
                icon = Icons.Default.Inventory,
                title = "Inventario",
                color = BioWayColors.InfoBlue,
                onClick = { /* TODO */ },
                modifier = Modifier.weight(1f)
            )
        }
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            MenuCard(
                icon = Icons.Default.Assessment,
                title = "Reportes",
                color = BioWayColors.OrangeAccent,
                onClick = { /* TODO */ },
                modifier = Modifier.weight(1f)
            )
            MenuCard(
                icon = Icons.Default.AccountBalance,
                title = "Prepago",
                color = BioWayColors.PurpleAccent,
                onClick = { /* TODO */ },
                modifier = Modifier.weight(1f)
            )
        }
    }
}

@Composable
private fun MenuCard(
    icon: ImageVector,
    title: String,
    color: Color,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier.clickable(onClick = onClick),
        shape = RoundedCornerShape(16.dp),
        color = Color.White,
        shadowElevation = 4.dp
    ) {
        Column(
            modifier = Modifier.padding(20.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(
                imageVector = icon,
                contentDescription = title,
                tint = color,
                modifier = Modifier.size(40.dp)
            )
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = title,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                color = BioWayColors.DarkGreen
            )
        }
    }
}

@Composable
private fun BuildRecentReceptions() {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        color = Color.White,
        shadowElevation = 4.dp
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            ReceptionItem("María García", "PET Tipo 1", 15.5, "$77.50", "10:30 AM")
            Divider(modifier = Modifier.padding(vertical = 12.dp))
            ReceptionItem("Juan Pérez", "Cartón", 25.0, "$50.00", "09:15 AM")
            Divider(modifier = Modifier.padding(vertical = 12.dp))
            ReceptionItem("Ana López", "Vidrio", 8.3, "$24.90", "08:45 AM")
        }
    }
}

@Composable
private fun ReceptionItem(
    name: String,
    material: String,
    kg: Double,
    amount: String,
    time: String
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = name,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                color = BioWayColors.DarkGreen
            )
            Text(
                text = "$material • $kg kg",
                fontSize = 14.sp,
                color = Color(0xFF666666)
            )
            Text(
                text = time,
                fontSize = 12.sp,
                color = Color(0xFF999999)
            )
        }
        Text(
            text = amount,
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = BioWayColors.SuccessGreen
        )
    }
}
